dmriprep/utils/vectors.py

```diff
@@ -375,3 +375,58 @@ def bvecs2ras(affine, bvecs, norm=True, bvec_norm_epsilon=0.2):
         rotated_bvecs[~b0s] /= norms_bvecs[~b0s, np.newaxis]
         rotated_bvecs[b0s] = np.zeros(3)
     return rotated_bvecs
+
+
+def _nonoverlapping_qspace_samples(
+    prediction_bval, prediction_bvec, all_bvals, all_bvecs, cutoff
+):
+    """Ensure that none of the training samples are too close to the sample to predict.
+    Parameters
```

Comment:
The top-level docstring should be made more general, to explain what this does and input Parameters could also be given more general names.